src/ExtendedSelenium2Library/__init__.py

```diff
@@ -584,3 +594,5 @@ def _wait_until_page_ready(self, timeout=None):
                 # instead of halting the process because document is not ready
                 # in <TIMEOUT>, we try our luck...
                 self._debug(exc_info()[0])
+            for keyword in self._page_ready_keyword_list:
+                BuiltIn().run_keyword(keyword)
```

Comment:
Can we instantiate BuiltIn once in __init__ . With current approach, the BuiltIn will get instantiated multiple times depending on the total number of items in the list.